rematbal/main.py

```diff
@@ -68,10 +72,9 @@ def fit_mbal_input(ts_obs, N, Wei, J):
             'dict_tank': self.tank_data
         }
 
-        ddi = [None] * len(self.prod_table['np'])
-        sdi = [None] * len(self.prod_table['np'])
-        wdi = [None] * len(self.prod_table['np'])
-        cdi = [None] * len(self.prod_table['np'])
```

Comment:
With this approach all 4 variable will be pointing to the same place in memory, if we assign a different value to the array ddi, this will affect variable sdi, wdi and cdi.

Reply:
Yes you are right! That is something I didn't know. Thank you!